rayon-core/src/registry.rs

```diff
@@ -159,23 +158,28 @@ pub(super) struct Registry {
 /// ////////////////////////////////////////////////////////////////////////
 /// Initialization
 
-static mut THE_REGISTRY: Option<&'static Arc<Registry>> = None;
+static mut THE_REGISTRY: Option<Arc<Registry>> = None;
 static THE_REGISTRY_SET: Once = Once::new();
 
 /// Starts the worker threads (if that has not already happened). If
 /// initialization has not already occurred, use the default
 /// configuration.
-fn global_registry() -> &'static Arc<Registry> {
+fn global_registry() -> Arc<Registry> {
     set_global_registry(|| Registry::new(ThreadPoolBuilder::new()))
-        .or_else(|err| unsafe { THE_REGISTRY.ok_or(err) })
+        .or_else(|err| unsafe {
+            THE_REGISTRY
+                .as_ref()
+                .map(|registry| registry.clone())
```

Comment:
I think we wanted to avoid the `Arc::clone` in cases that don't actually need an "owned" handle. It's potentially a performance issue to add those atomic count updates, although I don't think this is called in many places.

Another possibility is to replace `THE_REGISTRY` and `THE_REGISTRY_SET` with something like `THE_REGISTRY: once_cell::sync::OnceCell<Arc<Registry>>` to deal with initialization. Those accessors do return a reference too, which can be `&'static T` when coming from a `static` variable.

Reply:
I was hesistant to use `OnceCell` at first: I was afraid to introduce a new dependency. Since it's part of your proposals, I've updated the code to use `OnceCell` accordingly.

Reply:
`OnceCell` was generated an issue when testing. I've found another solution based on the first one, but without cloning the `Arc`.